nodebuilder/header/service.go

```diff
@@ -51,6 +52,33 @@ func (s *Service) GetVerifiedRangeByHeight(
 }
 
 func (s *Service) GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+	// TODO(vgonkivs): remove after https://github.com/celestiaorg/go-header/issues/32 will be
+	// implemented
+	head, err := s.store.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if uint64(head.Height()) < height {
+		return nil, fmt.Errorf("header: given height exceeds local head."+
+			"localHeadHeight:%d, requestedHeight:%d", head.Height(), height)
+	}
+	if uint64(head.Height()) == height {
+		return head, nil
+	}
+
+	head, err = s.syncer.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if uint64(head.Height()) < height {
```

Comment:
We should allow syncerHead+1 as well to accolade for a lag

Reply:
Not for now.

Reply:
I think here is the answer https://github.com/celestiaorg/celestia-node/pull/2173#discussion_r1189612682

Reply:
Synchead is not milliseconds - it could be 10 seconds and then a new synchead.Head request would have to be made to catch that header. So while we can do this with localHead, we cannot do this with synchead.